airflow/providers/http/hooks/http.py

```diff
@@ -66,7 +75,7 @@ def get_conn(self, headers=None):
             if conn.port:
                 self.base_url = self.base_url + ":" + str(conn.port)
             if conn.login:
-                session.auth = (conn.login, conn.password)
+                session.auth = self.auth_type(conn.login, conn.password)
```

Comment:
Other auth methods require a different number of arguments, e.g. OAuth2.What is the purpose of this change?

Reply:
the change make sense to me since it uses request package so it can work with custom auth
https://2.python-requests.org/en/master/user/advanced/#custom-authentication
provide the 
the auth_type in this context is auth object passed in the operator constructor

Reply:
https://requests.readthedocs.io/en/master/user/authentication/
Is it used with other methods?
I think it would be better if the user passed the auth object, not just the auth_type.

Reply:
@mik-laj the reason why I have given a default value is cus of backward compatibility.